git-branchless/src/revset/builtins.rs

```diff
@@ -398,3 +401,46 @@ fn fn_exactly(ctx: &mut Context, name: &str, args: &[Expr]) -> EvalResult {
         })
     }
 }
+
+fn fn_rewritten(ctx: &mut Context, name: &str, args: &[Expr]) -> EvalResult {
+    let expr = eval1(ctx, name, args)?;
+
+    let conn = ctx.repo.get_db_conn()?;
+    let event_log_db = EventLogDb::new(&conn)
+        .wrap_err("Connecting to event log")
+        .map_err(EvalError::OtherError)?;
+    let event_replayer = EventReplayer::from_event_log_db(ctx.effects, ctx.repo, &event_log_db)
+        .wrap_err("Retrieving event replayer")
+        .map_err(EvalError::OtherError)?;
+    let event_cursor = event_replayer.make_default_cursor();
+
+    let mut result = Vec::new();
+    for vertex in expr
+        .iter()
+        .wrap_err("Iterating commit set")
+        .map_err(EvalError::OtherError)?
+    {
+        let vertex = vertex
+            .wrap_err("Evaluating vertex")
+            .map_err(EvalError::OtherError)?;
+
+        let oid = NonZeroOid::try_from(vertex)
+            .wrap_err("Converting vertex to oid")
+            .map_err(EvalError::OtherError)?;
+
+        match find_rewrite_target(&event_replayer, event_cursor, oid) {
+            Some(new_commit_oid) => match new_commit_oid {
+                MaybeZeroOid::NonZero(new_commit_oid) => {
+                    // commit rewritten as new_commit_oid
+                    result.push(new_commit_oid);
+                }
+                MaybeZeroOid::Zero => {
+                    // commit deleted, skip
+                }
+            },
+            // commit not rewritten
+            None => result.push(oid),
```

Comment:
Is it accurate to call this function "rewritten" if it sometimes returns versions of the commits which haven't been rewritten?

Mercurial calls the concept itself `successors`:

> `successors(set)`
>     All successors for set, including the given set themselves

But I'm not really sure when you would want to get the successors of a set but also the set itself.

For most tasks, I think you would want either:

* For each commit, if it's been rewritten, produce the latest rewritten version, or else drop the the commit.
* For each commit, if it's been rewritten, produce the latest rewritten version, or else produce the commit itself (which is what you have here).

Maybe we should call this function `latest` instead?

Reply:
Ah, yes, thanks for pointing this out. I hadn't considered it, but I agree that `rewritten` is not accurate and could me misleading. I agree it should be changed, but I'm not *quite* set on `latest`, which feels like it could also mean "the commit in this set that was most recently created/commited/etc" as well as "latest version of the commits in this set". 

How would you feel about `current` instead of `latest`?

Reply:
Sure, `current` sounds fine to me!

Reply:
Great, thank you!